zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobTimeoutChecker.java

```diff
@@ -7,52 +7,29 @@
  */
 package io.camunda.zeebe.engine.processing.job;
 
-import static io.camunda.zeebe.scheduler.clock.ActorClock.currentTimeMillis;
-
 import io.camunda.zeebe.engine.state.immutable.JobState;
-import io.camunda.zeebe.engine.state.immutable.JobState.DeadlineIndex;
-import io.camunda.zeebe.protocol.record.intent.JobIntent;
 import io.camunda.zeebe.stream.api.ReadonlyStreamProcessorContext;
 import io.camunda.zeebe.stream.api.StreamProcessorLifecycleAware;
-import io.camunda.zeebe.stream.api.scheduling.Task;
-import io.camunda.zeebe.stream.api.scheduling.TaskResult;
-import io.camunda.zeebe.stream.api.scheduling.TaskResultBuilder;
 import java.time.Duration;
-import org.agrona.collections.MutableInteger;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public final class JobTimeoutChecker implements StreamProcessorLifecycleAware {
```

Comment:
🔧 As this class now cares about scheduling the checker triggered by the stream processor lifecycle, wdyt about naming it `JobTimeoutCheckerScheduler`? 🤔

Reply:
I was suggesting `JobTimeoutListner` or `JobTimeoutObserver` on a comment above, but `JobTimeoutCheckerScheduler` seems reasonable too!